### ISSUE TITLE
(Feature request) Add eslint

### ISSUE THREAD

**Issue by Myzel394** [zadam/trilium#2324]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/zadam/trilium/issues) for a feature request that matches the one I want to file, without success.

### Describe feature

A linter helps us having consistent code design

**Comment by Myzel394** [comment]
@zadam what do you think?

**Comment by zadam** [comment]
I'm fine with it as an advisory tool, but not as a pre commit hook, auto formatter etc.

Also of course depends on the rule set.

**Comment by Myzel394** [comment]
> I'm fine with it as an advisory tool, but not as a pre commit hook, auto formatter etc.

Why not? I think that would help us having a consistent codebase. For example: Double quotes or single quotes? Currently we use both kind of, which is a bit messy. With an auto formatter, we'd have it consistent.

**Comment by zadam** [comment]
It's a bikeshedding most of the time. "consistency" is in itself very vague goal without a clear practical benefit.

**Comment by Myzel394** [comment]
> It's a bikeshedding most of the time

I agree with small projects, but for such a large app like Trilium Notes, I think it would be very beneficial.

> "consistency" is in itself very vague goal without a clear practical benefit.

The benefit is that we will need less cognitive power to read the code because everything is formatted in a way we understand. This way, we can focus on the actual implementations and their problems, instead of trying to understand the code before we can actually implement it.
